### PR TITLE
[Ready for Review] Update the BinderHub Workshop document

### LINKED ISSUE #529: Improvements to zero-to-binderhub workshop
## Summary

After going through the instructions of the workshop [Zero-to-BinderHub](https://github.com/alan-turing-institute/the-turing-way/blob/master/workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md), I have compiled a series of comments. Some are just suggested improvements, others are typos/errors, others are things that might not work as expected, depending on the platform/software used:

### Suggestions
While the workshop is not intended to be stand-alone material, it will be useful to give a little bit of background information, in particular:
- [x] why it might be useful to create your own BinderHub rather than using [mybinder.org](https://mybinder.org), for example. What's the use case for creating your own BinderHub?
- [x] a high level description on what is to be done to have this up and running, i.e., the big picture of the following sections. This is not given even in the [BinderHub Docs](https://binderhub.readthedocs.io/en/latest/index.html), except in one brief sentence. What's Kubernetes used for? Why is it needed? And Docker? And JupyterHub (this is the only obvious one)? How do these things relate to each other?

### Things that might not work as expected
[DEPLOYING KUBERNETES CLUSTER ON AZURE](https://github.com/alan-turing-institute/the-turing-way/blob/master/workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md#deploying-a-kubernetes-cluster-on-azure)
- [x] STEP 1: `az login --output none` might not actually log in if you don't have a subscription, even if you successfully go through the login process via de Browser (tested with Safari). No idea why. My workaround for this was doing the login in the CL with `az login -u your_user_name` and then writing your password. 
- [x] STEP 5: This step seems to create two other different resource clusters that the one created in STEP 3. After following these two steps to the letter, I have in my Azure `testhub`, `NetworkWatcherRG` and `MC_testhub_hubcluster_westeurope` resource clusters.

[TEARING DOWN YOUR BINDERHUB DEPLOYMENT
](https://github.com/alan-turing-institute/the-turing-way/blob/master/workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md#tearing-down-your-binderhub-deployment)
- [x] STEP 3: As there are more resource groups than expected, one of them, `NetworkWatcherRG` is not deleted in this step.

### Errors/Typos
[SETUP BINDERHUB](https://github.com/alan-turing-institute/the-turing-way/blob/master/workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md#setup-binderhub)
- [x] STEP 2: The commented line `"s/<jupyter-ip>/${jupyter_ip}/"` should go AFTER the `config-template.yaml > config.yaml`, otherwise the `sed` command hangs. In STEP 4, after setting the IP, that line, uncommented, has to be put back before the config-template line. 
 
### Who can help?

* @sgibson91

---

### Updates

*Lorem ipsum dolor sit amet, consectetur adipiscing.*

### PR DESCRIPTION
**Pull request by sgibson91**
### Summary

This PR addresses the issues raised at the Reactor Sprint around the clarity of the Zero-to-BinderHub workshop.

Fixes #529 

### List of changes proposed in this PR (pull-request)

* Add alternative login command
* (Try to) Explain where extra resource groups come from when deploying the Kubernetes cluster
* Remind users to delete extraneous resource groups when tearing down the deployment
* Edit `config-template.yaml` to prevent `setup.sh` from hanging during execution due to a misplaced comment
* Add clearer steps for editing `setup.sh` and `config-template.yaml` when parsing the JupyterHub IP

### What should a reviewer concentrate their feedback on?

- [ ] Everything looks ok?

### Acknowledging contributors

- [x] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file.

### CODE REVIEW

**Comment by sgibson91** [comment]
@dalonsoa would you like to review?

**Comment by sgibson91** [comment]
Thank you!

@KirstieJane do you have magic admin privileges to merge despite broken tests?

File: workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md [hunk 1]
```diff
@@ -293,6 +350,8 @@ kubectl patch deployment tiller-deploy \
     }]'
 ```
 
+**NOTE:** I would recommend removing the line breaks and whitespace from this command.
```
Comment (LouiseABowler) [inline workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md]: Could you add why? If it's something people should do, is it worth providing a copy-pastable version of the command as well?

File: workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md [hunk 2]
```diff
@@ -293,6 +350,8 @@ kubectl patch deployment tiller-deploy \
...
+
 :question: 🤔
```
Comment (LouiseABowler) [inline workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md]: The thinking face looks a little like a to-do. The question mark one from above is good - could you add a similar comment here?

File: workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md [hunk 3]
```diff
@@ -436,7 +500,7 @@ You would probably create your own logo!
 
 #### 1. Fork the following repo
 
-I have created a template repo to make this easier linked below.
+I have created a template repo to make this easier - linked below.
 Fork it into your own namespace.
 
 [**github.com/sgibson91/binderhub-custom-files**](https://github.com/sgibson91/binderhub-custom-files)
```
Comment (LouiseABowler) [inline workshops/build-a-binderhub/workshop-presentations/zero-to-binderhub.md]: Should these files be in a repo under the Turing GitHub organisation?

**Pull request merged**